addons/pos_dpopay/models/dpopay_request.py

```diff
@@ -0,0 +1,118 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import logging
+import requests
+
+from requests.exceptions import (
+    ConnectionError,
+    HTTPError,
+    JSONDecodeError,
+    RequestException,
+    Timeout,
+)
+from odoo import _
+from odoo.exceptions import UserError
+
+_logger = logging.getLogger(__name__)
+
+DPOPAY_TOKEN_EXPIRED_CODES = ('999912', '999913')
+DPOPAY_CREDENTIAL_ERROR_CODES = ('401', '999911', '999960')
+DPOPAY_PROD_HOST = 'api.network.global'
+DPOPAY_TEST_HOST = 'api-dev.network.global'
+
+
+def _get_timeout(env):
+    try:
+        config_param = env['ir.config_parameter'].sudo()
+        value = config_param.get_param('pos_dpopay.request_timeout')
+        if not value:
+            config_param.set_param('pos_dpopay.request_timeout', 35)
+            return 35
+        return int(value)
+    except (ValueError, TypeError):
+        return 35
+
+
+def _get_dpopay_base_url(payment_method, is_token=False):
+    host = (payment_method.dpopay_test_mode and DPOPAY_TEST_HOST) or DPOPAY_PROD_HOST
+
+    if is_token:
+        return f'https://{host}/v1'
+    return f'https://{host}/ngenius-webapi/payments/push/v1/tid:{payment_method.dpopay_tid}/mid:{payment_method.dpopay_mid}'
+
+
+def _dpopay_headers(token, payment_method):
+    return {'Authorization': f'Bearer {token}', 'Chain-ID': payment_method.dpopay_header_value}
+
+
+def _refresh_dpopay_token(payment_method):
+    session = requests.Session()
+    auth = requests.auth.HTTPBasicAuth(payment_method.dpopay_client_id, payment_method.dpopay_client_secret)
+    url = f'{_get_dpopay_base_url(payment_method, True)}/tokenkc/generate'
+
+    response = session.get(url, auth=auth, timeout=_get_timeout(payment_method.env))
+    response.raise_for_status()
+    access_token = response.json().get('access_token')
+
+    if not access_token:
+        raise UserError(_('Unable to retrieve DPO Pay bearer token: check Client ID and Client Secret.'))
+
+    payment_method.dpopay_bearer_token = access_token
+    return _dpopay_headers(access_token, payment_method)
+
+
+def execute_dpopay_api_request(payment_method, payload, endpoint, action):
+    session = requests.Session()
+    timeout = _get_timeout(payment_method.env)
+    mode = 'Test' if payment_method.dpopay_test_mode else 'Production'
+    session.headers.update(_dpopay_headers(payment_method.dpopay_bearer_token, payment_method))
```

Comment:
```suggestion
    timeout = _get_timeout(payment_method.env)
    mode = 'Test' if payment_method.dpopay_test_mode else 'Production'
```
for header make variable and pass it in the request